### PR TITLE
fix(sql): fix UnsupportedOperationException on group by geohash column

### PR DESCRIPTION
**Pull request by bziobrowski**
This PR fixes UnsupportedOperationException for queries that try to group by geohash .

java.lang.UnsupportedOperationException
        at io.questdb.griffin.engine.functions.BinFunction.getGeoByte(BinFunction.java:139)
        at io.questdb.cairo.sql.VirtualRecord.getGeoByte(VirtualRecord.java:169)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.putGeoHashStringByteValue(JsonQueryProcessorState.java:342)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.doQueryRecord(JsonQueryProcessorState.java:529)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.doRecordFetchLoop(JsonQueryProcessorState.java:601)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.doFirstRecordLoop(JsonQueryProcessorState.java:412)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.onQueryPrefix(JsonQueryProcessorState.java:734)
        at io.questdb.cutlass.http.processors.JsonQueryProcessorState.resume(JsonQueryProcessorState.java:789)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.doResumeSend(JsonQueryProcessor.java:257)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.executeSelect(JsonQueryProcessor.java:449)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.executeNewSelect(JsonQueryProcessor.java:434)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.compileQuery(JsonQueryProcessor.java:356)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.execute0(JsonQueryProcessor.java:165)
        at io.questdb.cutlass.http.processors.JsonQueryProcessor.onRequestComplete(JsonQueryProcessor.java:217)
        at io.questdb.cutlass.http.HttpConnectionContext.handleClientRecv(HttpConnectionContext.java:568)
        at io.questdb.cutlass.http.HttpConnectionContext.handleClientOperation(HttpConnectionContext.java:246)
        at io.questdb.cutlass.http.HttpServer$1.lambda$$0(HttpServer.java:88)
        at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:166)
        at io.questdb.cutlass.http.HttpServer$1.run(HttpServer.java:92)
        at io.questdb.mp.Worker.run(Worker.java:113)

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByUtils.java|0|8|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByUtils.java|8|8|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/groupby/GroupByUtils.java [hunk 1]
```diff
@@ -176,6 +176,18 @@ public static void prepareGroupByRecordFunctions(
                         case ColumnType.LONG256:
                             fun = Long256Column.newInstance(keyColumnIndex - 1);
                             break;
+                        case ColumnType.GEOBYTE:
+                            fun = new GeoByteColumn(keyColumnIndex - 1, type);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/groupby/GroupByUtils.java:180]: could we implement factory method on `Geo*Column` classes, similar to what we have here:

image

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 70 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByUtils.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoIntColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoByteColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoShortColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoLongColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/constants/Constants.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 70 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByUtils.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoIntColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoByteColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoShortColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoLongColumn.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/constants/Constants.java|6|6|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/constants/Constants.java [hunk 2]
```diff
@@ -41,6 +41,14 @@ public static ConstantFunction getGeoHashConstant(long hash, int bits) {
...
     public static ConstantFunction getNullConstant(int columnType) {
+        int typeTag = ColumnType.tagOf(columnType);
+        if (typeTag >= ColumnType.GEOBYTE &&
+                typeTag <= ColumnType.GEOLONG) {
+            int bits = ColumnType.getGeoHashBits(columnType);
+            if (bits != 0) {
+                return geoNullConstants.get(bits);
+            }
+        }
         return nullConstants.getQuick(ColumnType.tagOf(columnType));
```
Comment (marregui) [inline core/src/main/java/io/questdb/griffin/engine/functions/constants/Constants.java:52]: could use `typeTag` and not call `ColumnType.tagOf(columnType)`

**Review by marregui — COMMENTED** [review]
knit picking :)

**Pull request merged**